### ISSUE TITLE
Notebook-less run's?

### ISSUE THREAD

**Issue by aersam** [fabricks-framework/fabricks#21]
Not sure if it's a good idea, but we could just not use notebooks for jobs that are not notebook's themselves. Notebooks are quite a fat thing, the imply a new spark session, new python session and therefore I guess we could save some time / compute. Not sure how much this adds up

**Comment by dominikpeter** [comment]
how woud you do it? If SQL run it in a thread?

**Comment by aersam** [comment]
The notebook API is also sync, so I guess there are threads already

**Comment by poskinf** [comment]
I see one big trade-off - I think notebooks make it easier to debug and to fix errors ( especially with the new ui) ... no ?

**Comment by aersam** [comment]
> I see one big trade-off - I think notebooks make it easier to debug and to fix errors ( especially with the new ui) ... no ?

Yes and no. Sure, the UI is nice. But: notebooks make programmatic access to the real exception impossible. For us internally, I mainly use the Trello and there we could just display the real exception (traceback.format_exception) instead of the link. This might be even faster for resolving it, we save a click :)

**Comment by aersam** [comment]
Trying to calculate if it's worth it: we have 2200 jobs which might take 3s in addition for the notebooks ^= 110min. I know we parallelize a lot and not all jobs can run without notebook, but I still think we might win around 20min or so with little effort
